### PR TITLE
ENG 2458 give all history rows a timestamp

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
All execution statuses:
- RegisteredExecutionStatus ExecutionStatus = "registered" -- Used by workflows, not operators
- PendingExecutionStatus ExecutionStatus = "pending"
- RunningExecutionStatus ExecutionStatus = "running" -- Doesn't seem to be an operator status. Noted here: https://linear.app/aqueducthq/issue/ENG-2626/currently-running-node-marked-as-pending
- CanceledExecutionStatus ExecutionStatus = "canceled"
- FailedExecutionStatus ExecutionStatus = "failed"
- SucceededExecutionStatus ExecutionStatus = "succeeded"
- UnknownExecutionStatus ExecutionStatus = "unknown"

Tested execution statuses on operators: pending, canceled, failed, succeeded. Skipped unknown because can fail for many reasons.
All have an associated timestamp except for pending execution status.

Fix: Add pending timestamp on creation & use pending_at time if status == pending. For unknown, use the latest available time. Otherwise, "Unknown".

## Related issue number (if any)
ENG 2458

## Demo (if any)
Before:
image

After:
Screen Shot 2023-03-15 at 12 36 35 PM

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [N/A] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/workflows/artifact/metric/history.tsx [hunk 1]
```diff
@@ -52,10 +52,31 @@ const MetricsHistory: React.FC<Props> = ({ historyWithLoadingStatus }) => {
         let timestamp = new Date(
```
Comment (likawind) [inline src/ui/common/src/components/workflows/artifact/metric/history.tsx:52]: Does the following makes sense:
* take the max of `all_times` using `max(all_times.map(x => new Date(x)))`. We could figure out if there's any method in `Date` object to tell if the date is valid, and filter out any invalid one.
* render 'Unknown' if none timestamp above is available, and render the string version if the above block produce a valid date.
This should allow us to get rid of most branching in this block

File: src/ui/common/src/components/workflows/artifact/check/history.tsx [hunk 2]
```diff
@@ -51,14 +51,21 @@ const CheckHistory: React.FC<CheckHistoryProps> = ({
...
-          timestamp = 'Unknown';
-        }
+        const timesOrNull = all_times.map((x) =>
+          typeof x === 'string' ? new Date(x) : null
+        );
+
+        const maxTime = Math.max.apply(null, timesOrNull);
+
+        const timestamp =
+          maxTime > 0 ? new Date(maxTime).toLocaleString() : 'Unknown';
```
Comment (likawind) [inline src/ui/common/src/components/workflows/artifact/check/history.tsx:68]: If we following the suggested change in line 61, we can assign this value based on if the list is empty

File: src/ui/common/src/components/workflows/artifact/check/history.tsx [hunk 3]
```diff
@@ -51,14 +51,21 @@ const CheckHistory: React.FC<CheckHistoryProps> = ({
...
 
-        // Checks that are canceled / fail to execute have no exec_state or finished_at time.
-        if (timestamp === 'Invalid Date') {
-          timestamp = 'Unknown';
-        }
+        const timesOrNull = all_times.map((x) =>
+          typeof x === 'string' ? new Date(x) : null
+        );
+
+        const maxTime = Math.max.apply(null, timesOrNull);
```
[the remark this answers is not in the document]
Comment (eunice-chan) [reply src/ui/common/src/components/workflows/artifact/check/history.tsx:65]: I tried removing it and date got rendered as a big number (Unix time, I think). My understanding is the first new Date(x)  converts from a string to a date object. Math.max converts it to a number and the second new Date converts from Unix time to the date object which I can apply toLocaleString() to

[the remark this answers is not in the document]
Comment (eunice-chan) [reply src/ui/common/src/components/workflows/artifact/check/history.tsx:65]: Added as a comment

**Review by likawind — APPROVED** [review]
:shipit: Thanks for addressing all comments!

**Pull request merged**